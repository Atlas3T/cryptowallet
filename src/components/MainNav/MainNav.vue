<template>
  <div class="columns is-mobile main-nav is-gapless">
    <div class="column">
      <router-link
        class="main-nav-link"
        to="/balance"
      >
        <q-icon name="fas fa-wallet" />
        <div>Balance</div>
      </router-link>
    </div>
    <div class="column">
      <router-link
        class="main-nav-link"
        to="/analytics"
      >
        <q-icon name="fas fa-chart-bar" />
        <div>Analytics</div>
      </router-link>
    </div>
    <div class="column">
      <router-link
        class="main-nav-link"
        to="/payments"
      >
        <q-icon name="fas fa-exchange-alt" />
        <div>Payments</div>
      </router-link>
    </div>
    <div class="column">
      <router-link
        class="main-nav-link"
        to="/exchange"
      >
        <q-icon name="fas fa-sync" />
        <div>Exchange</div>
      </router-link>
    </div>
    <div class="column">
      <router-link
        class="main-nav-link"
        to="/more"
      >
        <q-icon name="fas fa-ellipsis-h" />
        <div>More</div>
      </router-link>

    </div>
  </div>
</template>

<script>
export default {
  name: 'MainNav',
};
</script>

<style scoped>
  .main-nav {
    width: 100%;
    height: 15vh;
  }

  .main-nav > .column {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 5px;
  }

  .main-nav-link {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: black;
  }

  .main-nav-link i {
    margin-top: 0.3rem;
    font-size: 2.5rem;
  }

  .main-nav-link > div {
    margin-top: 0.5rem;
    font-size: 0.8rem;
  }
</style>
