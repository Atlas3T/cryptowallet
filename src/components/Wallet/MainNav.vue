<template>
  <section
    class="main-nav"
  >
    <div class="">
      <router-link
        class="main-nav-link"
        to="/wallet/wallet"
      >
        <i class="svg-icon">
          <img src="~assets/icons/Wallet.svg">
        </i>
        <div>{{ $t('wallet') }}</div>
      </router-link>
    </div>
    <div class="">
      <router-link
        class="main-nav-link"
        to="/wallet/analytics"
      >
        <i class="svg-icon">
          <img src="~assets/icons/Analytics.svg">
        </i>
        <div>{{ $t('analytics') }}</div>
      </router-link>
    </div>
    <div class="">
      <router-link
        class="main-nav-link"
        to="/wallet/payments"
      >
        <i class="svg-icon">
          <img src="~assets/icons/Payments.svg">
        </i>
        <div>{{ $t('payments') }}</div>
      </router-link>
    </div>
    <div class="">
      <router-link
        class="main-nav-link"
        to="/wallet/exchange"
      >
        <i class="svg-icon">
          <img src="~assets/icons/Exchange.svg">
        </i>
        <div>{{ $t('exchange') }}</div>
      </router-link>
    </div>
    <div class="">
      <router-link
        class="main-nav-link"
        to="/wallet/more"
      >
        <i class="svg-icon">
          <img src="~assets/icons/More.svg">
        </i>
        <div>{{ $t('more') }}</div>
      </router-link>

    </div>
  </section>
</template>

<script>
export default {
  name: 'MainNav',
  data() {
    return {
      icon: '/assets/icons/Analytics.svg',
    };
  },
};
</script>

<style scoped>
  .main-nav {
    width: 100%;
    height: 6rem;
    background: white;
    background: #f3f3f3;
    display: flex;
    justify-content: space-between;
  }

  .main-nav > div {
    flex: 1;
  }

  .main-nav-link {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: black;
    width: 100%;
    font-family: 'Montserrat-Regular';
    height: 100%;
  }

  .main-nav-link i {
    font-size: 2.5rem;
  }

  .main-nav-link > div {
    font-size: 0.8rem;
  }

  .router-link-active {
    background: #eaeaea;
  }

  .svg-icon {
    width: 100%;
    height: 4rem;
    line-height: 0;
    display: flex;
  }

  .svg-icon img {
    max-width: 75%;
    margin: 0 auto;
  }
</style>
