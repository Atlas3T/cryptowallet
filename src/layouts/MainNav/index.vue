<template>
  <section
    class="main-nav"
  >
    <div class="">
      <router-link
        class="main-nav-link"
        to="/wallet"
      >
        <q-icon name="account_balance_wallet" />
        <div>{{ $t('wallets') }}</div>
      </router-link>
    </div>
    <div class="">
      <router-link
        class="main-nav-link"
        to="/settings"
      >
        <q-icon name="settings" />
        <div>{{ $t('settings') }}</div>
      </router-link>
    </div>
  </section>
</template>

<script>
export default {
  name: 'MainNav',
};
</script>

<style scoped>
  .main-nav {
    width: 100%;
    height: 5rem;
    background: #0a2d3e;
    display: flex;
    justify-content: space-between;
    border-top: 0.3rem solid #4e677d;
    text-shadow: 1px 1px 0px rgba(0, 0, 0, 0.5);
  }

  .main-nav > div {
    flex: 1;
  }

  .main-nav-link {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: white;
    width: 100%;
    font-family: 'Montserrat-Medium';
    height: 100%;
    text-decoration: none;
  }

  .main-nav-link i {
    font-size: 2em;
    margin-bottom: 0.2em;
  }

  .main-nav-link > div {
    font-size: 0.8rem;
  }

  .router-link-exact-active {
    color: #16ac9f;
    font-family: 'Montserrat-Medium';
  }

  .svg-icon {
    width: 100%;
    height: 4rem;
    line-height: 0;
    display: flex;
  }

  .svg-icon img {
    max-width: 75%;
    margin: 0 auto;
  }
</style>
