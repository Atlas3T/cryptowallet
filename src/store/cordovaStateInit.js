/*
 * this function will consume the complete Vuex store object and
 * add event listeners to Cordova's deviceready
 */
export default function (store) {
  /*
   * getDeviceContacts will attempt to get all of the devices contacts
   * as well as all of the fields available on each contact
   * if it succedes its going to mutate the Vuex store
   */
  function getDeviceContacts() {
    function onError() {
      throw new Error('getDeviceContacts(): Failed to retrieve contacts from the device.');
    }

    const options = new ContactFindOptions();
    options.multiple = true;
    const fields = ['*'];
    navigator.contacts.find(fields, (contacts) => {
      store.state.wallet.contacts = contacts;
    }, onError, options);
  }

  document.addEventListener('deviceready', () => {
    getDeviceContacts();
  }, false);
}
