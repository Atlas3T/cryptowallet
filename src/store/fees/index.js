import { Model } from '@vuex-orm/core';

export default class fees extends Model {
  static entity = 'fees';

  static primaryKey = ['code'];

  static fields() {
    return {
      code: this.attr(''),
      timestamp: this.attr(''),
      data: this.attr(''),
    };
  }
}
