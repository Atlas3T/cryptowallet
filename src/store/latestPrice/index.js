import { Model } from '@vuex-orm/core';


export default class latestPrice extends Model {
    static entity = 'latestPrice';

    static primaryKey = ['coin', 'currency']


    static fields() {
      return {
        coin: this.attr(''),
        currency: this.attr(''),
        updated: this.attr(''),
        data: this.attr(''),
      };
    }
}
