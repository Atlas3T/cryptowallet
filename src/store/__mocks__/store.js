import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
