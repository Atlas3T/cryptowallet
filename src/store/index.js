/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
