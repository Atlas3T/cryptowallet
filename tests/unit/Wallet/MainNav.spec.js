import { createLocalVue, mount } from '@vue/test-utils';
import MainNav from '@/components/Wallet/MainNav.vue';
import { __createMocks as createStoreMocks } from '@/../tests/store/__mocks__/store.js';
import { localVue, router } from '../setupLocalVue';

const menuItems = ['Balance', 'Analytics', 'Payments', 'Exchange', 'More'];

describe('MainNav.vue', () => {
  let storeMocks;
  let wrapper;
  let store;

  function wrapperInit (options) {
    return mount(MainNav, options);
  }

  function storeInit (custom) {
    storeMocks = createStoreMocks(custom);
    wrapper = wrapperInit({ router, localVue, store: storeMocks.store, });
    store = wrapper.vm.$store;
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a section with .main-nav class', () => {
    expect(wrapper.contains('section.main-nav')).toBe(true);
  });

  it('renders all of the menu items', () => {
    expect(wrapper.findAll('a.main-nav-link').length).toBe(5);
    menuItems.forEach((item, index) => {
      expect(wrapper.findAll('a').at(index).text()).toBe(menuItems[index]);
    });
  });

  it('updates the path on each menu item click', () => {
    wrapper.findAll('a').at(0).trigger('click');
    expect(wrapper.vm.$route.path).toBe('/wallet/balance');
    wrapper.findAll('a').at(1).trigger('click');
    expect(wrapper.vm.$route.path).toBe('/wallet/analytics');
    wrapper.findAll('a').at(2).trigger('click');
    expect(wrapper.vm.$route.path).toBe('/wallet/payments');
    wrapper.findAll('a').at(3).trigger('click');
    expect(wrapper.vm.$route.path).toBe('/wallet/exchange');
    wrapper.findAll('a').at(4).trigger('click');
    expect(wrapper.vm.$route.path).toBe('/wallet/more');
  });

  it('renders a correct icon on each menu item', () => {
    expect(wrapper.findAll('a').at(0).contains('.fa-wallet')).toBe(true);
    expect(wrapper.findAll('a').at(1).contains('.fa-chart-bar')).toBe(true);
    expect(wrapper.findAll('a').at(2).contains('.fa-exchange-alt')).toBe(true);
    expect(wrapper.findAll('a').at(3).contains('.fa-sync')).toBe(true);
    expect(wrapper.findAll('a').at(4).contains('.fa-ellipsis-h')).toBe(true);
  });
});
