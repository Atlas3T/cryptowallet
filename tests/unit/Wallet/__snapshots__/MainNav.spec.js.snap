// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav.vue renders and matches snapshot 1`] = `
<section
  class="columns is-mobile main-nav is-gapless"
>
  <div
    class="column"
  >
    <a
      class="main-nav-link"
      href="#/wallet/balance"
    >
      <i
        aria-hidden="true"
        class="q-icon fas fa-wallet"
      >
         
      </i>
       
      <div>
        Balance
      </div>
    </a>
  </div>
   
  <div
    class="column"
  >
    <a
      class="main-nav-link"
      href="#/wallet/analytics"
    >
      <i
        aria-hidden="true"
        class="q-icon fas fa-chart-bar"
      >
         
      </i>
       
      <div>
        Analytics
      </div>
    </a>
  </div>
   
  <div
    class="column"
  >
    <a
      class="main-nav-link"
      href="#/wallet/payments"
    >
      <i
        aria-hidden="true"
        class="q-icon fas fa-exchange-alt"
      >
         
      </i>
       
      <div>
        Payments
      </div>
    </a>
  </div>
   
  <div
    class="column"
  >
    <a
      class="main-nav-link"
      href="#/wallet/exchange"
    >
      <i
        aria-hidden="true"
        class="q-icon fas fa-sync"
      >
         
      </i>
       
      <div>
        Exchange
      </div>
    </a>
  </div>
   
  <div
    class="column"
  >
    <a
      class="main-nav-link"
      href="#/wallet/more"
    >
      <i
        aria-hidden="true"
        class="q-icon fas fa-ellipsis-h"
      >
         
      </i>
       
      <div>
        More
      </div>
    </a>
  </div>
</section>
`;
